Numerics/EvaluatingFunctions/remez.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This python script implements Remez' algorithm.
+# To use the script, you can modify what happens when this script is executed:
+# See the `if __name__ == "__main__":` branch near the end of this file.
+#
+# Alternatively, `runRemez` can be imported and run directly, see the docstring
+# of runRemez.
+
+import math
+import numpy as np
+from matplotlib import pyplot as plt
+
+# Return n chebyshev nodes on the interval (a,b)
+def _getChebyshevNodes(n, a, b):
+    nodes = [.5*(a+b) + .5*(b-a)*math.cos((2*k+1)/(2.*n)*math.pi) for k in range(n)]
+    return nodes
+
+# Return the error on given nodes of a polynomial with coefficients polycoeff
+# approximating the function with function values exactvals (on these nodes).
+def _getErrors(exactvals, polycoeff, nodes):
+    ys = np.polyval(polycoeff, nodes)
+    for i in range(len(ys)):
+        ys[i] = abs(ys[i]-exactvals[i])
+    return ys
+
+"""
+Return the coefficients of a polynomial of degree d approximating
+the function fun on the interval (a,b).
+
+Args:
+    fun: Function to approximate
+    a: Left interval border
+    b: Right interval border
+    d: The polynomial degree will be d, 2*d or 2*d + 1 depending
+        on the values of odd and even below
+    odd: If True, use odd polynomial of degree 2*d+1
+    even: If True, use even polynomial of degree 2*d
+    tol: Tolerance to use when checking for convergence
+
+Returns: Tuple where the first entry is the achieved absolute error
+    and the second entry is a list of the polynomial coefficients in
+    the order that is required by the QDK Numerics library. This is
+    the inverse order compared to what np.polyval expects.
+"""
+def runRemez(fun, a, b, d=5, odd=False, even=False, tol=1.e-13):
+    finished = False
+    # initial set of points for the interpolation
+    cn = getChebyshevNodes(d + 2, a, b)
+    # mesh on which we'll evaluate the error
+    cn2 = getChebyshevNodes(100 * d, a, b)
+
+    # do at most 50 iterations and cancel if we "lose" an interpolation
+    # point
+    it = 0
+    while not finished and len(cn)==d+2 and it < 50:
+        it += 1
+        # set up the linear system of equations for Remez' algorithm
+        b = np.array([fun(c) for c in cn])
+        A = np.matrix(np.zeros([d+2,d+2]))
+        for i in range(d+2):
+            x = 1.
+            if odd:
+                x *= cn[i]
+            for j in range(d+2):
+                A[i, j] = x
+                x *= cn[i]
+                if odd or even:
+                    x *= cn[i]
+            A[i, -1] = (-1)**(i+1)
+        # this will give us a polynomial interpolation
+        res = np.linalg.solve(A, b)
+
+        # add padding for even/odd polynomials
+        revlist = list(reversed(res[0:-1]))
```

Comment:
It looks like `revlist` is only used in the `for` loop on L78, such that capturing it in a list isn't necessary?